src/main/kotlin/mynameisjeff/skyblockclientupdater/utils/ExpectedAndReceived.kt

```diff
@@ -0,0 +1,80 @@
+package mynameisjeff.skyblockclientupdater.utils
+import java.lang.Math;
+
+object ExpectedAndReceived {
+    val expected: String? = null // string a
+    val received: String? = null // string b
+
+    constructor(expected: String, received: String) {
+        this.expected = expected
+        this.received = received
+    }
+
+    fun needsUpdate(distance: Integer): Boolean {
+        val exemptChars = Array<Char?>(6) ['_', '-', '+', ' ', '.', '!']
+        val numbers = Array<Char?>(10) ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
+
+        var expectedChars = ArrayList<Char>()
+        var receivedChars = ArrayList<Char>()
+
+        /*
+        for index in range(0, max(len(stra), len(strb))):
+            if index < min(len(stra), len(strb)):
+                if stra[index] != strb[index]:
+                    print(f"Letter Expected: {stra[index]}")
+                    print(f"Letter Recieved: {strb[index]}")
+                    print()
+            else:
+                a = "null"
+                b = "null"
+                if len(stra) > len(strb):
+                    a = stra[index]
+                else:
+                    b = strb[index]
+
+                print(f"Letter Expected: {a}" )
+                print(f"Letter Recieved: {b}")
+                print()
+        */
+
+        for (index in 0..Math.max(expected.length(), received.length())) {
```

Comment:
Could probbaly use a filter function instead but it's fine if it works

Reply:
no clue how they work